### ISSUE TITLE
Connection to server got closed. Server will not be restarted

### ISSUE THREAD

**Issue by harshil21** [microsoft/vscode-cpptools#9427]
Bug type: Language Service

**Describe the bug**
- OS and Version: Windows 11, using WSL2 - ubuntu 22.04: Output of `uname -a`: `Linux [omitted] 5.10.102.1-microsoft-standard-WSL2 #1 SMP Wed Mar 2 00:30:59 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux`
- VS Code Version: 1.67.2
- C/C++ Extension Version: 1.10.4 pre-release. But also affects 1.9.8 stable
- Other extensions you installed (and if the issue persists after disabling them): Github Copilot, Gitlens, Pylance, Python
- If using SSH remote, specify OS of remote machine: -
- A clear and concise description of what the bug is, including information about the workspace (i.e. is the workspace a single project or multiple projects, size of the project, etc).

I'm finally following up on #8871, the same issue is still happening, details attached in that issue are still relevant. To answer the questions in [this](https://github.com/microsoft/vscode-cpptools/issues/8871#issuecomment-1040692342) comment:

> Are you able to create a workspace folder with a .vscode/c_cpp_properties.json file but no C/C++ files open and then attach to cpptools, and then open a file to cause the crash to get a call stack?

Not quite, I closed my .cpp file, then re-launched vscode. The language server crashed all 5 times in 3 minutes. I re-opened vscode again. This time it crashed 3 times. The 4th time it worked. I attached the debugger and opened the `hello_world.cpp` file. It did not crash anymore, but the inspection doesn't work right (see the picture attached below). Here is what my workspace and files look like:

.vscode
    - c_cpp_properties.json
    - settings.json
    - launch.json
    - tasks.json
hello_world.cpp

Content of each of these files:

<details> 
<summary> launch.json: </summary>

```
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "(gdb) Attach",
            "type": "cppdbg",
            "request": "attach",
            "processId": "${command:pickProcess}",
            "program": "/home/hoppingturtles/.vscode-server/extensions/ms-vscode.cpptools-1.10.4-linux-x64/bin/cpptools",
            "MIMode": "gdb",
            "setupCommands": [
                {
                    "description": "Enable pretty-printing for gdb",
                    "text": "-enable-pretty-printing",
                    "ignoreFailures": true
                }
            ]
        }
    ]
}
```

</details>

<details>
<summary> settings.json </summary>

```
{
    "C_Cpp.loggingLevel": "Debug"
}
```

</details>

<details>
<summary> tasks.json </summary>

```
{
  "version": "2.0.0",
  "tasks": [
    {
      "type": "shell",
      "label": "g++ build active file",
      "command": "/usr/bin/g++",
      "args": [
        "-g",
        "${file}",
        "-o",
        "${fileDirname}/${fileBasenameNoExtension}"
      ],
      "options": {
        "cwd": "/usr/bin"
      },
      "problemMatcher": [
        "$gcc"
      ],
      "group": {
        "kind": "build",
        "isDefault": true
      }
    }
  ]
}
```

</details>

<details>
<summary> c_cpp_properties.json </summary>

```
{
  "version": "2.0.0",
  "tasks": [
    {
      "type": "shell",
      "label": "g++ build active file",
      "command": "/usr/bin/g++",
      "args": [
        "-g",
        "${file}",
        "-o",
        "${fileDirname}/${fileBasenameNoExtension}"
      ],
      "options": {
        "cwd": "/usr/bin"
      },
      "problemMatcher": [
        "$gcc"
      ],
      "group": {
        "kind": "build",
        "isDefault": true
      }
    }
  ]
}
```

</details>

<details>
<summary> hello_world.cpp </summary>

```
#include <iostream>
#include <vector>
#include <string>

using namespace std;

int main()
{
   vector<string> msg {"Hello", "C++", "World", "from", "VS Code", "and the C++ extension!"};

   for (const string& word : msg)
   {
      cout << word << " ";
   }
   cout << endl;
}
```

</details>

> Also, you may want to try uninstalling the extension and re-installing it

Nope, did not work. Switching to the latest pre-release did not either.

**Steps to reproduce**

1. Open VSCode workspace with above files
2. 

**Expected behavior**

#### Code sample and logs 

see above

- Logs from running `C/C++: Log Diagnostics` from the VS Code command palette

<details>
</summary>Logs</summary>

```
-------- Diagnostics - 6/8/2022, 10:48:58 PM
Version: 1.10.4
Current Configuration:
{
    "name": "Linux",
    "includePath": [
        "${workspaceFolder}/**"
    ],
    "defines": [],
    "compilerPath": "/usr/bin/gcc",
    "cStandard": "c11",
    "cppStandard": "c++17",
    "intelliSenseMode": "clang-x64",
    "compilerPathIsExplicit": true,
    "cStandardIsExplicit": true,
    "cppStandardIsExplicit": true,
    "intelliSenseModeIsExplicit": true,
    "compilerArgs": [
        "--std",
        "c++20"
    ],
    "mergeConfigurations": false,
    "browse": {
        "path": [
            "${workspaceFolder}/**"
        ],
        "limitSymbolsToIncludedHeaders": true
    }
}
No active translation units.

------- Workspace parsing diagnostics -------
Number of files discovered (not excluded): 6032
```

</details>

- Logs from [the language server logging](https://code.visualstudio.com/docs/cpp/enable-logging-cpp#_enable-logging-for-the-language-server)

<details>
<summary> C/C++ Debug logs - This includes me restarting vscode twice after closing the .cpp file </summary>

```
loggingLevel: Debug
cpptools/didChangeCppProperties
Attempting to get defaults from C compiler in "compilerPath" property: '/usr/bin/gcc'
Querying compiler for default C++ language standard using command line: /usr/bin/gcc -x c++ -E -dM /dev/null
Detected language standard version: gnu++17
Querying compiler for default C language standard using command line: /usr/bin/gcc -x c -E -dM /dev/null
Detected language standard version: gnu17
Querying compiler's default target using command line: "/usr/bin/gcc" -dumpmachine
Compiler returned default target value: x86_64-linux-gnu
cpptools/didChangeCppProperties
Attempting to get defaults from C compiler in "compilerPath" property: '/usr/bin/gcc'
Querying compiler for default C++ language standard using command line: /usr/bin/gcc -x c++ -E -dM /dev/null
Detected language standard version: gnu++17
Querying compiler for default C language standard using command line: /usr/bin/gcc -x c -E -dM /dev/null
Detected language standard version: gnu17
Querying compiler's default target using command line: "/usr/bin/gcc" -dumpmachine
Compiler returned default target value: x86_64-linux-gnu
cpptools/didChangeCppProperties
Attempting to get defaults from C compiler in "compilerPath" property: '/usr/bin/gcc'
Querying compiler for default C++ language standard using command line: /usr/bin/gcc -x c++ -E -dM /dev/null
Detected language standard version: gnu++17
Querying compiler for default C language standard using command line: /usr/bin/gcc -x c -E -dM /dev/null
Detected language standard version: gnu17
Querying compiler's default target using command line: "/usr/bin/gcc" -dumpmachine
Compiler returned default target value: x86_64-linux-gnu
cpptools/didChangeCppProperties
Attempting to get defaults from C compiler in "compilerPath" property: '/usr/bin/gcc'
Querying compiler for default C++ language standard using command line: /usr/bin/gcc -x c++ -E -dM /dev/null
Detected language standard version: gnu++17
Querying compiler for default C language standard using command line: /usr/bin/gcc -x c -E -dM /dev/null
Detected language standard version: gnu17
Querying compiler's default target using command line: "/usr/bin/gcc" -dumpmachine
Compiler returned default target value: x86_64-linux-gnu
Compiler query command line: /usr/bin/gcc 'REE 2Failed to query compiler. Falling back to 32-bit intelliSenseMode.
Compiler query command line: /usr/bin/gcc 'REE 2Failed to query compiler. Falling back to no bitness.
Compiler query command line: /usr/bin/gcc 'REE 2ignoring nonexistent directory "/usr/local/include/x86_64-linux-gnu"
ignoring nonexistent directory "/usr/lib/gcc/x86_64-linux-gnu/11/include-fixed"
ignoring nonexistent directory "/usr/lib/gcc/x86_64-linux-gnu/11/../../../../x86_64-linux-gnu/include"
#include "..." search starts here:
#include <...> search starts here:
 /usr/lib/gcc/x86_64-linux-gnu/11/include
 /usr/local/include
 /usr/include/x86_64-linux-gnu
 /usr/include
End of search list.
gcc: warning: REE 2: linker input file unused because linking not done
gcc: error: REE 2: linker input file not found: No such file or directory

Code browsing service initialized
Attempting to get defaults from C++ compiler in "compilerPath" property: '/usr/bin/gcc'
Compiler query command line: /usr/bin/gcc '?Failed to query compiler. Falling back to 32-bit intelliSenseMode.
Compiler query command line: /usr/bin/gcc '?Failed to query compiler. Falling back to no bitness.
Compiler query command line: /usr/bin/gcc '?ignoring duplicate directory "/usr/include/x86_64-linux-gnu/c++/11"
ignoring nonexistent directory "/usr/local/include/x86_64-linux-gnu"
ignoring nonexistent directory "/usr/lib/gcc/x86_64-linux-gnu/11/include-fixed"
ignoring nonexistent directory "/usr/lib/gcc/x86_64-linux-gnu/11/../../../../x86_64-linux-gnu/include"
#include "..." search starts here:
#include <...> search starts here:
 /usr/include/c++/11
 /usr/include/x86_64-linux-gnu/c++/11
 /usr/include/c++/11/backward
 /usr/lib/gcc/x86_64-linux-gnu/11/include
 /usr/local/include
 /usr/include/x86_64-linux-gnu
 /usr/include
End of search list.
gcc: warning: ?: linker input file unused because linking not done
gcc: error: ?: linker input file not found: No such file or directory

  Folder: /usr/include/ will be indexed
  Folder: /usr/lib/llvm-14/lib/clang/14.0.0/include/ will be indexed
  Folder: /usr/local/include/ will be indexed
  Folder: /mnt/c/Users/Harshil Mehta/Documents/cpp-projects/ will be indexed
cpptools/didChangeSettings
Discovering files...
IntelliSense Engine = Default.
Enhanced Colorization is enabled.
Error squiggles are enabled if all header dependencies are resolved.
Autocomplete is enabled.
  Processing folder (recursive): /usr/include/
  Processing folder (recursive): /usr/lib/llvm-14/lib/clang/14.0.0/include/
  Processing folder (recursive): /usr/local/include/
  Processing folder (recursive): /mnt/c/Users/Harshil Mehta/Documents/cpp-projects/
  Discovering files: 6032 file(s) processed
  0 file(s) removed from database
Done discovering files.
Populating include completion cache.
Parsing remaining files...
  Parsing: 0 files(s) processed
Done parsing remaining files.
```
</details>

**Screenshots**

![image](https://user-images.githubusercontent.com/37377066/172694688-813a3c41-fad8-4338-9850-725d935798e0.png)

**Comment by sean-mcmanus** [comment]
Your c_cpp_properties.json is invalid. Did you copy/paste it wrong? If not, can you delete that file and regenerate it via one of the Edit Configuration commands?

**Comment by bobbrow** [comment]
I think it's a copy/paste error.  The Log Diagnostics shows something that may be the culprit.

```
    "cppStandard": "c++17",
    "intelliSenseMode": "clang-x64",
    "compilerPathIsExplicit": true,
    "cStandardIsExplicit": true,
    "cppStandardIsExplicit": true,
    "intelliSenseModeIsExplicit": true,
    "compilerArgs": [
        "--std",
        "c++20"
    ],
```

Try removing `compilerArgs` from `c_cpp_properties.json` and setting `cppStandard` to `"c++20"`

**Comment by harshil21** [comment]
Thanks for the quick reply!

I don't remember where I got that from, but deleting and regenerating it doesn't work (because the server crashed), so I modified my `c_cpp_properties.json` to:
```
{
  "configurations": [
    {
      "name": "Linux",
      "includePath": ["${workspaceFolder}/**"],
      "defines": [],
      "compilerPath": "/usr/bin/gcc",
      "cStandard": "c11",
      "cppStandard": "c++20",
      "intelliSenseMode": "${default}"
    }
  ],
  "version": 4
}
```

And sadly it's still not fixed, here are the language server logs:

<details>
<summary> Logs </summary>

```
loggingLevel: Debug
cpptools/didChangeCppProperties
Attempting to get defaults from C compiler in "compilerPath" property: '/usr/bin/gcc'
Querying compiler for default C++ language standard using command line: /usr/bin/gcc -x c++ -E -dM /dev/null
Detected language standard version: gnu++17
Querying compiler for default C language standard using command line: /usr/bin/gcc -x c -E -dM /dev/null
Detected language standard version: gnu17
Querying compiler's default target using command line: "/usr/bin/gcc" -dumpmachine
Compiler returned default target value: x86_64-linux-gnu
Compiler query command line: /usr/bin/gcc 'es":0}}Failed to query compiler. Falling back to 32-bit intelliSenseMode.
Compiler query command line: /usr/bin/gcc 'es":0}}Failed to query compiler. Falling back to no bitness.
Compiler query command line: /usr/bin/gcc 'es":0}}ignoring nonexistent directory "/usr/local/include/x86_64-linux-gnu"
ignoring nonexistent directory "/usr/lib/gcc/x86_64-linux-gnu/11/include-fixed"
ignoring nonexistent directory "/usr/lib/gcc/x86_64-linux-gnu/11/../../../../x86_64-linux-gnu/include"
#include "..." search starts here:
#include <...> search starts here:
 /usr/lib/gcc/x86_64-linux-gnu/11/include
 /usr/local/include
 /usr/include/x86_64-linux-gnu
 /usr/include
End of search list.
gcc: warning: es":0}}: linker input file unused because linking not done
gcc: error: es":0}}: linker input file not found: No such file or directory

Code browsing service initialized
Attempting to get defaults from C++ compiler in "compilerPath" property: '/usr/bin/gcc'
cpptools/didChangeCppProperties
Attempting to get defaults from C compiler in "compilerPath" property: '/usr/bin/gcc'
Querying compiler for default C++ language standard using command line: /usr/bin/gcc -x c++ -E -dM /dev/null
Detected language standard version: gnu++17
Querying compiler for default C language standard using command line: /usr/bin/gcc -x c -E -dM /dev/null
Detected language standard version: gnu17
Querying compiler's default target using command line: "/usr/bin/gcc" -dumpmachine
Compiler returned default target value: x86_64-linux-gnu
cpptools/didChangeCppProperties
Attempting to get defaults from C compiler in "compilerPath" property: '/usr/bin/gcc'
Querying compiler for default C++ language standard using command line: /usr/bin/gcc -x c++ -E -dM /dev/null
Detected language standard version: gnu++17
Querying compiler for default C language standard using command line: /usr/bin/gcc -x c -E -dM /dev/null
Detected language standard version: gnu17
Querying compiler's default target using command line: "/usr/bin/gcc" -dumpmachine
Compiler returned default target value: x86_64-linux-gnu
cpptools/didChangeCppProperties
Attempting to get defaults from C compiler in "compilerPath" property: '/usr/bin/gcc'
Querying compiler for default C++ language standard using command line: /usr/bin/gcc -x c++ -E -dM /dev/null
Detected language standard version: gnu++17
Querying compiler for default C language standard using command line: /usr/bin/gcc -x c -E -dM /dev/null
Detected language standard version: gnu17
Querying compiler's default target using command line: "/usr/bin/gcc" -dumpmachine
Compiler returned default target value: x86_64-linux-gnu
Compiler query command line: /usr/bin/gcc '0Failed to query compiler. Falling back to 32-bit intelliSenseMode.
Compiler query command line: /usr/bin/gcc '0Failed to query compiler. Falling back to no bitness.
Compiler query command line: /usr/bin/gcc '0ignoring nonexistent directory "/usr/local/include/x86_64-linux-gnu"
ignoring nonexistent directory "/usr/lib/gcc/x86_64-linux-gnu/11/include-fixed"
ignoring nonexistent directory "/usr/lib/gcc/x86_64-linux-gnu/11/../../../../x86_64-linux-gnu/include"
#include "..." search starts here:
#include <...> search starts here:
 /usr/lib/gcc/x86_64-linux-gnu/11/include
 /usr/local/include
 /usr/include/x86_64-linux-gnu
 /usr/include
End of search list.
gcc: warning: 0: linker input file unused because linking not done
gcc: error: 0: linker input file not found: No such file or directory

Code browsing service initialized
Attempting to get defaults from C++ compiler in "compilerPath" property: '/usr/bin/gcc'
cpptools/didChangeCppProperties
Attempting to get defaults from C compiler in "compilerPath" property: '/usr/bin/gcc'
Querying compiler for default C++ language standard using command line: /usr/bin/gcc -x c++ -E -dM /dev/null
Detected language standard version: gnu++17
Querying compiler for default C language standard using command line: /usr/bin/gcc -x c -E -dM /dev/null
Detected language standard version: gnu17
Querying compiler's default target using command line: "/usr/bin/gcc" -dumpmachine
Compiler returned default target value: x86_64-linux-gnu
Compiler query command line: /usr/bin/gcc 't defaults from C compiler in \"compilerPath\" proper' -std=c11 -m64 -Wp,-v -E -dM -x c /dev/null
Failed to query compiler. Falling back to 32-bit intelliSenseMode.
Compiler query command line: /usr/bin/gcc 't defaults from C compiler in \"compilerPath\" proper' -std=c11 -m32 -Wp,-v -E -dM -x c /dev/null
Failed to query compiler. Falling back to no bitness.
Compiler query command line: /usr/bin/gcc 't defaults from C compiler in \"compilerPath\" proper' -std=c11 -Wp,-v -E -dM -x c /dev/null
ignoring nonexistent directory "/usr/local/include/x86_64-linux-gnu"
ignoring nonexistent directory "/usr/lib/gcc/x86_64-linux-gnu/11/include-fixed"
ignoring nonexistent directory "/usr/lib/gcc/x86_64-linux-gnu/11/../../../../x86_64-linux-gnu/include"
#include "..." search starts here:
#include <...> search starts here:
 /usr/lib/gcc/x86_64-linux-gnu/11/include
 /usr/local/include
 /usr/include/x86_64-linux-gnu
 /usr/include
End of search list.
gcc: warning: t defaults from C compiler in \"compilerPath\" proper: linker input file unused because linking not done
gcc: error: t defaults from C compiler in \"compilerPath\" proper: linker input file not found: No such file or directory

Code browsing service initialized
Attempting to get defaults from C++ compiler in "compilerPath" property: '/usr/bin/gcc'
```

</details>

something to do with finding the file it seems.

**Comment by harshil21** [comment]
I do, running `which g++` outputs: `/usr/bin/g++`. But changing it to that doesn't seem to make any difference, still getting the same error. Now here's the language server logs when I restart VSCode for the 2nd time:

<details>
<summary> Logs </summary>

```
loggingLevel: Debug
cpptools/didChangeCppProperties
Attempting to get defaults from C compiler in "compilerPath" property: '/usr/bin/g++'
Querying compiler for default C++ language standard using command line: /usr/bin/g++ -x c++ -E -dM /dev/null
Detected language standard version: gnu++17
Querying compiler for default C language standard using command line: /usr/bin/g++ -x c -E -dM /dev/null
Detected language standard version: gnu17
Querying compiler's default target using command line: "/usr/bin/g++" -dumpmachine
Compiler returned default target value: x86_64-linux-gnu
cpptools/didChangeCppProperties
Attempting to get defaults from C compiler in "compilerPath" property: '/usr/bin/g++'
Querying compiler for default C++ language standard using command line: /usr/bin/g++ -x c++ -E -dM /dev/null
Detected language standard version: gnu++17
Querying compiler for default C language standard using command line: /usr/bin/g++ -x c -E -dM /dev/null
Detected language standard version: gnu17
Querying compiler's default target using command line: "/usr/bin/g++" -dumpmachine
Compiler returned default target value: x86_64-linux-gnu
cpptools/didChangeCppProperties
Attempting to get defaults from C compiler in "compilerPath" property: '/usr/bin/g++'
Querying compiler for default C++ language standard using command line: /usr/bin/g++ -x c++ -E -dM /dev/null
Detected language standard version: gnu++17
Querying compiler for default C language standard using command line: /usr/bin/g++ -x c -E -dM /dev/null
Detected language standard version: gnu17
Querying compiler's default target using command line: "/usr/bin/g++" -dumpmachine
Compiler returned default target value: x86_64-linux-gnu
cpptools/didChangeCppProperties
Attempting to get defaults from C compiler in "compilerPath" property: '/usr/bin/g++'
Querying compiler for default C++ language standard using command line: /usr/bin/g++ -x c++ -E -dM /dev/null
Detected language standard version: gnu++17
Querying compiler for default C language standard using command line: /usr/bin/g++ -x c -E -dM /dev/null
Detected language standard version: gnu17
Querying compiler's default target using command line: "/usr/bin/g++" -dumpmachine
Compiler returned default target value: x86_64-linux-gnu
Compiler query command line: /usr/bin/g++ 'es":0}}Failed to query compiler. Falling back to 32-bit intelliSenseMode.
Compiler query command line: /usr/bin/g++ 'es":0}}Failed to query compiler. Falling back to no bitness.
Compiler query command line: /usr/bin/g++ 'es":0}}ignoring nonexistent directory "/usr/local/include/x86_64-linux-gnu"
ignoring nonexistent directory "/usr/lib/gcc/x86_64-linux-gnu/11/include-fixed"
ignoring nonexistent directory "/usr/lib/gcc/x86_64-linux-gnu/11/../../../../x86_64-linux-gnu/include"
#include "..." search starts here:
#include <...> search starts here:
 /usr/lib/gcc/x86_64-linux-gnu/11/include
 /usr/local/include
 /usr/include/x86_64-linux-gnu
 /usr/include
End of search list.
g++: warning: es":0}}: linker input file unused because linking not done
g++: error: es":0}}: linker input file not found: No such file or directory

Code browsing service initialized
Attempting to get defaults from C++ compiler in "compilerPath" property: '/usr/bin/g++'
Compiler query command line: /usr/bin/g++ (���Failed to query compiler. Falling back to 32-bit intelliSenseMode.
Compiler query command line: /usr/bin/g++ (���Failed to query compiler. Falling back to no bitness.
Compiler query command line: /usr/bin/g++ (���ignoring duplicate directory "/usr/include/x86_64-linux-gnu/c++/11"
ignoring nonexistent directory "/usr/local/include/x86_64-linux-gnu"
ignoring nonexistent directory "/usr/lib/gcc/x86_64-linux-gnu/11/include-fixed"
ignoring nonexistent directory "/usr/lib/gcc/x86_64-linux-gnu/11/../../../../x86_64-linux-gnu/include"
#include "..." search starts here:
#include <...> search starts here:
 /usr/include/c++/11
 /usr/include/x86_64-linux-gnu/c++/11
 /usr/include/c++/11/backward
 /usr/lib/gcc/x86_64-linux-gnu/11/include
 /usr/local/include
 /usr/include/x86_64-linux-gnu
 /usr/include
End of search list.
g++: warning: (���: linker input file unused because linking not done
g++: error: (���: linker input file not found: No such file or directory

  Folder: /usr/include/ will be indexed
  Folder: /usr/lib/llvm-14/lib/clang/14.0.0/include/ will be indexed
  Folder: /usr/local/include/ will be indexed
  Folder: /mnt/c/Users/Harshil Mehta/Documents/cpp-projects/ will be indexed
cpptools/didChangeSettings
IntelliSense Engine = Default.
Enhanced Colorization is enabled.
Error squiggles are enabled if all header dependencies are resolved.
Discovering files...
Autocomplete is enabled.
  Processing folder (recursive): /usr/include/
  Processing folder (recursive): /usr/lib/llvm-14/lib/clang/14.0.0/include/
  Processing folder (recursive): /usr/local/include/
  Processing folder (recursive): /mnt/c/Users/Harshil Mehta/Documents/cpp-projects/
  Discovering files: 6033 file(s) processed
  0 file(s) removed from database
Done discovering files.
Populating include completion cache.
Parsing remaining files...
  Parsing: 0 files(s) processed
Done parsing remaining files.
```

</details>

I think now i'll just create a new workspace and follow the steps in https://code.visualstudio.com/docs/cpp/config-wsl#_cc-configurations again

**Comment by sean-mcmanus** [comment]
I don't know where the `'es":0}}` in your log is coming from. Looks like some garbage data.

@Colengms Could this be related to the create process changes?

**Comment by Colengms** [comment]
There seems to be some sort of memory corruption occurring, corrupting the arguments passed to the compiler probe.  For example:

```
Compiler query command line: /usr/bin/gcc 't defaults from C compiler in \"compilerPath\" proper' -std=c11 -m64 -Wp,-v -E -dM -x c /dev/null
Failed to query compiler. Falling back to 32-bit intelliSenseMode.
Compiler query command line: /usr/bin/gcc 't defaults from C compiler in \"compilerPath\" proper' -std=c11 -m32 -Wp,-v -E -dM -x c /dev/null
Failed to query compiler. Falling back to no bitness.
Compiler query command line: /usr/bin/gcc 't defaults from C compiler in \"compilerPath\" proper' -std=c11 -Wp,-v -E -dM -x c /dev/null
```

In this case, the first argument passed to gcc appears to be a bit of log output (`t defaults from C compiler in \"compilerPath\" proper`), which was 'properly' quoted and escaped before being passed as an argument.  So, the issue appears to be prior to the process launching logic, and seems to be related to compiler argument handling.  That bad argument appears to have been processed as if provided by the configuration.  The issue appears to only occur when applying arguments to the compiler query that come from the configuration.  The simple queries for default target and standard version, appear to be working properly.

If we could get a repro of this, it may be relatively simple to address.  But, this is the first report I'm aware of, and I'm not able to repro it locally.  There may be something specific to the repro environment that we may need to uncover to figure out what's going on.

@harshil21 Does this issue also occur if you move your project to a native Linux directory instead of being mounting from Windows on `/mnt/c`?  Perhaps move it to `~/` instead?  Perhaps there is some issue with reading `c_cpp_properties.json` via the mount point.

**Comment by harshil21** [comment]
> Does this issue also occur if you move your project to a native Linux directory instead of being mounting from Windows on `/mnt/c`? Perhaps move it to `~/` instead? Perhaps there is some issue with reading `c_cpp_properties.json` via the mount point.

I just moved my hello world file to `~/` and unfortunately the error is still happening.

I also simplified my setup, deleting the rest of the files .vscode folder and only keeping `settings.json` for the logging level. These are the logs:

<details>
<summary> C/C++ logs </summary>

```
loggingLevel: Debug
cpptools/didChangeCppProperties
Attempting to get defaults from C compiler in "compilerPath" property: '/usr/bin/clang'
Compiler query command line: /usr/bin/clang P��Failed to query compiler. Falling back to 32-bit intelliSenseMode.
Compiler query command line: /usr/bin/clang P��Failed to query compiler. Falling back to no bitness.
Compiler query command line: /usr/bin/clang P��clang: error: no such file or directory: 'P��[0;1;36m'

Code browsing service initialized

```

<details>

Seems like it still leaking something into the arguments

**Comment by Colengms** [comment]
Hi @harshil21 .  Did you create your Ubuntu 22.02 WSL2 instance using "Ubuntu 22.04 LTS" from the Microsoft Store?  Could you try creating a fresh instance using that, and only running the following:

```
sudo apt update
sudo apt upgrade
sudo apt install gcc g++
```

... Then connecting to it from VS Code (using the `New WSL Window using Distro...` command), adding the C/C++ Extension, opening a new empty folder in your home directory, and adding a test.cpp file?  Doing that myself, I see that gcc is successfully queried.  I'm curious if this works for you, as that might isolate the issue to something specific to your WSL instance.

**Comment by harshil21** [comment]
> Did you create your Ubuntu 22.04 WSL2 instance using "Ubuntu 22.04 LTS"

hey so I had installed `Ubuntu` from the Microsoft store which at the time had installed `20.04` on my machine (Windows 10 at the time). I had upgraded it by running `sudo apt dist-upgrade` shortly after 22.04 was released.

I just installed Ubuntu 22.04 from the Microsoft store and followed the rest of the instructions, and everything seems to be working now!

I think I'll just migrate to this WSL instance since my old one couldn't run graphical apps either and this one had a entire graphical UI setup (wow) and debugging the older instance is probably gonna suck anyway.

Thanks for the resolve! Closing.
